src/ui/shared/requests/useEventSource.ts

```diff
@@ -14,6 +14,7 @@ interface Options<T> {
   enabled?: boolean;
   mapResponse?: (response: unknown) => T;
   mergeResponse?(currentValue: T | null, nextValue: T | null): T | null;
+  onEnd?: (value: null | T) => void;
```

Comment:
What about `onDone`?

Reply:
That way my first though as well, but ended up naming it `onEnd` to match the existing event name and the corresponding event handler